examples/quad/main.rs

```diff
@@ -137,663 +138,749 @@ fn main() {
-
-    let img = image::load(Cursor::new(&img_data[..]), image::PNG)
-        .unwrap()
-        .to_rgba();
-    let (width, height) = img.dimensions();
-    let kind = i::Kind::D2(width as i::Size, height as i::Size, 1, 1);
-    let row_alignment_mask = limits.optimal_buffer_copy_pitch_alignment as u32 - 1;
-    let image_stride = 4usize;
-    let row_pitch = (width * image_stride as u32 + row_alignment_mask) & !row_alignment_mask;
-    let upload_size = (height * row_pitch) as u64;
-
-    let mut image_upload_buffer =
-        unsafe { device.create_buffer(upload_size, buffer::Usage::TRANSFER_SRC) }.unwrap();
-    let image_mem_reqs = unsafe { device.get_buffer_requirements(&image_upload_buffer) };
-    let image_upload_memory =
-        unsafe { device.allocate_memory(upload_type, image_mem_reqs.size) }.unwrap();
-
-    unsafe { device.bind_buffer_memory(&image_upload_memory, 0, &mut image_upload_buffer) }
-        .unwrap();
-
-    // copy image data into staging buffer
-    unsafe {
-        let mut data = device
-            .acquire_mapping_writer::<u8>(&image_upload_memory, 0 .. image_mem_reqs.size)
-            .unwrap();
-        for y in 0 .. height as usize {
-            let row = &(*img)
-                [y * (width as usize) * image_stride .. (y + 1) * (width as usize) * image_stride];
-            let dest_base = y * row_pitch as usize;
-            data[dest_base .. dest_base + row.len()].copy_from_slice(row);
+        match event {
+            winit::event::Event::WindowEvent { event, .. } => {
+                match event {
+                    winit::event::WindowEvent::CloseRequested => *control_flow = winit::event_loop::ControlFlow::Exit,
+                    winit::event::WindowEvent::KeyboardInput {
+                        input:
+                            winit::event::KeyboardInput {
+                                virtual_keycode: Some(winit::event::VirtualKeyCode::Escape),
+                                ..
+                            },
+                        ..
+                    } => *control_flow = winit::event_loop::ControlFlow::Exit,
+                    winit::event::WindowEvent::Resized(dims) => {
+                        println!("resized to {:?}", dims);
+                        #[cfg(feature = "gl")]
+                        {
+                            let context = renderer.surface.get_context();
+                            context.resize(dims.to_physical(context.window().hidpi_factor()));
+                        }
+                        let dimensions = Extent2D {
+                            width: dims.width as u32,
+                            height: dims.height as u32,
+                        };
+                        renderer.dimensions = dimensions;
+                        renderer.recreate_swapchain();
+                    }
+                    _ => { }
+                }
+            }
+            winit::event::Event::EventsCleared => {
+                renderer.render();
```

Comment:
You should be calling winit's `request_redraw` here.

`renderer_state.draw();` should be moved to `RedrawRequested`, so that it also redraws when the OS requests a redraw.

Reply:
https://github.com/rust-windowing/winit/issues/1051 This issue is clashing really badly with RedrawRequested